src/test/compile-fail/lint-exceeding-bitshifts.rs

```diff
@@ -11,7 +11,7 @@
 #![deny(exceeding_bitshifts)]
 #![allow(unused_variables)]
 #![allow(dead_code)]
-#![feature(num_bits_bytes)]
+#![feature(num_bits_bytes, const_indexing)]
```

Comment:
Hmm.  Sorry, I was not clear about what I was saying.

I was worried about the fact that you added the feature gate opt-in, since this very bug (that this PR is addressing) is about the fact that people shouldn't have to add this feature gate to get this code to work.

But it seems like you are saying that you need the feature gate there in order to see the compile-time error on *just* the line that you added, and without the feature gate, there would be no compile-time error -- i.e. the lint wouldn't detect the excessive bitshift at compile-time.

I would have preferred such a subtle issue be broken out into a separate test.  But its not worth blocking the PR at this point.